engine/xml-config/src/main/java/org/datacleaner/configuration/JaxbConfigurationReader.java

```diff
@@ -52,56 +52,16 @@
 import org.apache.metamodel.util.SimpleTableDef;
 import org.apache.metamodel.xml.XmlSaxTableDef;
 import org.datacleaner.api.RenderingFormat;
-import org.datacleaner.configuration.jaxb.AbstractDatastoreType;
-import org.datacleaner.configuration.jaxb.AccessDatastoreType;
-import org.datacleaner.configuration.jaxb.BerkeleyDbStorageProviderType;
-import org.datacleaner.configuration.jaxb.CassandraDatastoreType;
-import org.datacleaner.configuration.jaxb.ClasspathScannerType;
+import org.datacleaner.configuration.jaxb.*;
```

Comment:
[cid:image001.png@01D0EEC4.D3B2B750]

Eclipse has this too, of course. See the two settings in the lower right corner of the dialogue above. This means that by default eclipse needs a 100 imports from the same package before it will automatically provide you with a wildcard (*) import statement. Personally I think it’s safer to rarely (if ever) use this wild card, because, although it’s cleaner, it’s more prone to lead to class name conflicts.

Menno.

From: Kasper Sørensen [mailto:notifications@github.com]
Sent: Thursday, September 10, 2015 20:33
To: datacleaner/DataCleaner
Subject: Re: [DataCleaner] Feature/653 remote component provider (#654)


In engine/xml-config/src/main/java/org/datacleaner/configuration/JaxbConfigurationReader.java<https://github.com/datacleaner/DataCleaner/pull/654#discussion_r39196200>:

> @@ -52,56 +52,16 @@

>  import org.apache.metamodel.util.SimpleTableDef;

>  import org.apache.metamodel.xml.XmlSaxTableDef;

>  import org.datacleaner.api.RenderingFormat;

> -import org.datacleaner.configuration.jaxb.AbstractDatastoreType;

> -import org.datacleaner.configuration.jaxb.AccessDatastoreType;

> -import org.datacleaner.configuration.jaxb.BerkeleyDbStorageProviderType;

> -import org.datacleaner.configuration.jaxb.CassandraDatastoreType;

> -import org.datacleaner.configuration.jaxb.ClasspathScannerType;

> +import org.datacleaner.configuration.jaxb.*;

While I do kind of "like" this different way of organizing imports, I fear that we will every time be committing back and forwards with new styles of organizing imports because we use different IDEs. I don't think eclipse has a way to do it the same way as IntelliJ does, so I guess I would like to ask you to revert to the Eclipse-way where every import is explicitly imported and

—
Reply to this email directly or view it on GitHub<https://github.com/datacleaner/DataCleaner/pull/654/files#r39196200>.